### PR TITLE
Video Bot + new events-based architecture

### LINKED ISSUE #756: Fix Video Bot
As [mentioned](https://joystream.notion.site/Impact-of-Olympia-on-Community-Tools-92e684153fa0484a84383a38203a2e70) on Notion, we need to bring the community tooling back to life. 

This ticket is about Video bot that announces all new Atlas videos in a dedicated discord channel. 

Estimate: `7-10h` 
Story Points: `8`
Priority: `Major`

### PR DESCRIPTION
**Pull request by singulart**
1. Fixes https://github.com/Joystream/community-repo/issues/756
2. Introduces new architectural change which may enable future testing

Essentially, with this change the node RPC connection is needed exactly once. Individual bots can subscribe to specific blockchain events in a nice and declarative way: `@OnEvent('*.LeaderSet')`

### CODE REVIEW

File: codegen.yml [hunk 1]
```diff
@@ -138,6 +140,296 @@ documents:
...
+    fragment DistributionBucketOperatorField on DistributionBucketOperator {
+      id
+      metadata {
+        nodeEndpoint
+        __typename
+      }
+      status
+      __typename
+    }
+
```
Comment (traumschule) [inline codegen.yml:432]: wonder if it's practical to outsource the fragments.

File: config.ts [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
 // video bot
 export const channelId = "938526399801729024";
-export const hydraLocation = "https://query.joystream.org/graphql";
-export const waitFor = 60;
-export const waitTimeUnit = "seconds";
-export const createdAgo = 30;
-export const createdAgoUnit = "minutes";
+export const pioneerApi = "https://query.joystream.org/graphql";
+export const atlasApi = "https://orion.joystream.org/graphql";
+//export const assetDistributorNode = "https://antonm.cryptostarter.info/distributor/api/v1/assets/";
```
Comment (freakstatic) [inline config.ts:12]: This comment can probably be removed

**Review by freakstatic — APPROVED** [review]
LGTM!
I wonder if there aren't going to sometimes happen problems related with a block being finalized and the QN being still syncing that block :thinking: 

The events based architecture looks good! makes the code quite clean

**Comment by singulart** [comment]
> LGTM! I wonder if there aren't going to sometimes happen problems related with a block being finalized and the QN being still syncing that block 🤔
> 
> The events based architecture looks good! makes the code quite clean

There is a retry mechanism with exponential back-off. See `globalRetryConfig` in `config.ts`

**Comment by singulart** [comment]
@DzhideX Please review this

**Review by DzhideX — APPROVED** [review]
Looks good on my end!

p.s. Sorry for the delay..

**Pull request merged**